retrieval/targetmanager_test.go

```diff
@@ -12,3 +12,63 @@
 // limitations under the License.
 
 package retrieval
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+	"gopkg.in/yaml.v2"
+
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/storage/local"
+)
+
+func TestTargetSetRecreatesTargetGroupsEveryRun(t *testing.T) {
+	scrapeConfig := &config.ScrapeConfig{}
+
+	sOne := `
+job_name: "foo"
+dns_sd_configs:
+- names:
+  - "srv.name.one.example.org"
+`
+	if err := yaml.Unmarshal([]byte(sOne), scrapeConfig); err != nil {
+		t.Fatalf("Unable to load YAML config sOne: %s", err)
+	}
```

Comment:
Hmm.  That's actually how I first wrote the test -- constructing the `map[string]TargetProvider` by hand, instead of parsing a string of YAML.  Unfortunately, I found it markedly harder to understand.  I had to set up a mockTargetProvider to implement that interface, and then muck around with Labels and LabelSets a bunch (which had to magically have an `AddressLabel` in order for the name to get found the right way).  It wasn't easy to see what the test was actually doing.

Once I figured out how to parse the YAML, it felt both easier to understand the test by itself, and also closer to the reported issue (I actually pasted those snippets from one of the GitHub issues).

All of which is to say: it's totally your call, and if you want me to go back to the manually-constructed objects, I'm happy to do so, but I wanted to make at least a brief pitch for sticking with the parsing of the YAML.  

So, I guess, lemme know what your instinct is.